src/Api.Core/PathHelper.cs

```diff
@@ -160,12 +160,43 @@ private static PathInformation ColloquialString2PathInformationInternal( [NotNul
 				path = DelimiterString + path;
 
 			// convert to database format by appending a delimiter if it is not already present (beware of escaping)
-			if( !path.EndsWith( DelimiterString, StringComparison.Ordinal ) || path.EndsWith( EscapedDelimiter, StringComparison.Ordinal ) )
+			if(!HasDelimiter( path ))
 				path += DelimiterString;
 
 			return String2PathInformationInternal( path.AsSpan(), "".AsSpan(), stringPool, entity );
 		}
 
+		/// <summary>
+		/// Checks whether the path string already contains a delimiter at the end.
+		/// </summary>
+		private static bool HasDelimiter( string path )
+		{
+			var isEscaped = false;
+			var hasDelimiter = false;
+
+			foreach( var character in path )
+			{
+				switch( character )
+				{
+					case Escape when !isEscaped:
+						isEscaped = true;
+						continue;

```

Comment:
Mhh wenn der Pfad auf "/\" endet (theoretisch keine legale Form, aber möglich), dann ergibt das hier true oder? Ist natürlich streitbar, was mit nem illegalen Pfad passieren soll, ich würde es aber bevorzugen, wenn in diesem Fall nicht nochmal ein / angehängt werden würde in Zeile 164. Reicht es vielleicht aus zu prüfen ob der Pfad auf einen / endet und davor nicht eine gerade Anzahl (inklusive 0) an \ ist?

Reply:
Was genau meinst du mit dem letzten Fall?
Ich habe das ganze mit dem Namen "BadName\" im Planner getestet und es wurde kein Slash als Teil des Namens hinzugefügt. Ich glaube, dass Escaping passiert an anderer Stelle und das Slash, was in Zeile 164 hinzugefügt wird, wird nicht als Teil des Namens interpretiert.

Reply:
Auch mit "BadName/\\" funktioniert es im Planner.

Reply:
Ich würde ja vermuten, dass der Planner Backslashes escaped, so dass diese Randfälle mittels Planner nicht entstehen sollten. Aber dann hätte das ursprüngliche Problem, dass der PR hier lösen sollte doch eigentlich auch schon nicht entstehen können sollen.
Wie dem auch sei ich habe mir nur den Code hier angeschaut und verglichen was die Methoden sagen, was sie tun und was sie tatsächlich tun. Wobei, nur Code angeschaut, sag mir wenn ich mich irre, könnte sein.

- "Foo/Bar/\\" -> Hier müsste eigentlich HasDelimiter zum Schluß kommen, dass es mit einem Trennzeichen endet. Das finde ich als Ergebnis der Methode überraschend, auch wenn es in Zeile 163 dann zum richtigen Verhalten führt. Die Verständlichkeit der HasDelemiter-Methode leidet da aber drunter.
- "Foo/Bar\\" -> Hier kommt die HasDelimiter-Methode zum Schluß, das es nicht mir einem Delimiter endet. Das sollte dann in Zeile 163 aber dazu führen, dass ein Delimiter angehangen wird. Nur wird das dann durch das schon vorhandene Backslash zu einem Pfadelementnamen, was doch sehr seltsames Verhalten ist.

Wenn das Fehlverhalten durch weitere vorherige Checks garnicht entstehen kann, dann kann man natürlich damit leben, wie diese Methode jetzt implementiert ist, ich wollte nur drauf hinweisen.

Reply:
"Foo/Bar/\" hat am Ende eben kein Trennzeichen. Für uns ist ja nur ein / das Pfadtrennzeichen. \ ist lediglich fürs Escaping. Darum sollte dort der / angefügt werden für die Datenbank. 
Bei "Foo/Bar\" müsste wie du sagst der / dran. Dann sieht es erstmal aus wie ein escapetes /, aber solange der Rückweg davon ausgeht, dass immer ein / am Ende ist, wird das dann einfach wieder entfernt. Slange der Vor- und Rückweg die gleichen Verfahren anwendet sollte der Roundtrip gelingen.

Reply:
Das eigentlich Problem hier war der einfache check per EndsWith. Vom Planner kommt schon ein korrekt escaptes Backslash und der nötige Slash als Trennzeichen: `\\/`. Nur hat es der PathHelper als escapetes Trennzeichen erkannt, wo noch ein / dran soll, statt zu merken "hier ist alles i.O.".

Reply:
Wir reden hier durchaus aneinander vorbei. Deinen ersten Punkt für "Foo/Bar/\\" verstehe ich schon, sehe ich genauso, der Code hier sagt aber aktuell, dass es sehr wohl auf Trennzeichen endet. Und ich vermute absichtlich, damit der Slash dann in 163 nicht angehängt wird. Das passt nun aber nicht zu deiner Roundtrip-Strategie oder?.

Reply:
Du hast recht, in dem Fall wäre das Ergebnis gewesen, dass Foo/Bar/\ auf einen Delimiter endet. Das war allerdings keine Absicht, sondern ich habe es übersehen, weil im eigentlichen Anwendungsfall daraus Foo\\/Bar\\/\\\\ geworden wäre, wo das nicht passiert. Ich habe die Methode nochmal angepasst.

Reply:
Ah ok, ich dachte du meinst mit _"Hier müsste eigentlich HasDelimiter zum Schluß kommen, dass es mit einem Trennzeichen endet."_ das es dein erwartetes Verhalten ist. Aber du hast Recht, das ist auch das Ergebnis der Methode. Das wäre dann aber wirklich inkorrekt. Der Code betrachtet bei `"Foo/Bar/\"` das vorletzte Slash als Trennzeichen und gibt deshalb true zurück, jedoch ist der angegebene String streng genommen nicht korrekt escapet, und endet eben nicht auf einem Trennzeichen. Die Frage ist, was passiert wenn wir wirklich stupide den Slash dann trotzdem dranhängen, was ja für das DB-Format so sein müsste und auf dem Rückweg wieder entfernt wird.

Reply:
Der Slash am Ende spielt bei dieser Eingabe keine Rolle. Egal ob mit oder ohne, Die Methode ```GetPathElementsFromQuotedString``` wirft den Fehler, "The path string contains invalid quoting." Aber das wäre ja das korrekte Verhalten, oder?

Reply:
Ja ich hätte es nicht im Konjunktiv schreiben sollen, das rührte noch daher, dass ich den Code nur gelesen aber nicht ausprobiert habe. Ich glaube dann haben wir hier alles geklärt.